sdk/keyvault/internal/challenge_policy.go

```diff
@@ -57,6 +58,10 @@ func (k *KeyVaultChallengePolicy) Do(req *policy.Request) (*http.Response, error
 			return nil, err
 		}
 
+		if resp.StatusCode > 299 {
```

Comment:
Does this look reasonable to you?

Reply:
I think this should go after the `err = k.findScopeAndTenant(resp)` line. The first response should be a 404 with the authorization in the `WWW-Authenticate` header.

Reply:
The problem here is that there's no `WWW-Authenticate` header in a failure response.  I noticed then when running a test against a key vault that had been deleted.  We kept retrying due to the missing header value.

Reply:
Can we extend this conditional to say `if resp.StatusCode > 299 && _, ok := resp.Header.Get("WWW-Authenticate"); !ok {...}` ? We don't want to return the error all the time, only when we have no info from the service and the status code is > 299

Reply:
Not sure I follow.  Also, I think the condition should probably be > 399